### ISSUE TITLE
Galata: more generic `waitForApplication` for other JupyterLab distributions

### ISSUE THREAD

**Issue by jtpio** [jupyterlab/jupyterlab#11414]
### Problem

Currently Galata waits for a couple of things to happen in JupyterLab as part of the `waitForApplication` logic:

https://github.com/jupyterlab/jupyterlab/blob/114159269a76c4a8d4c1e725e882d7c8a7791869/galata/src/fixtures.ts#L277-L295

However in other lab-based distributions such as RetroLab, `#jupyterlab-splash` does not exist and there is no launcher.

### Proposed Solution

For now it's possible to override that with:

```ts
import { test as base } from '@jupyterlab/galata';

export const test = base.extend({
  waitForApplication: async ({ baseURL }, use, testInfo) => {
    const waitIsReady = async (page): Promise<void> => {
      await page.waitForSelector('#main-panel');
    };
    await use(waitIsReady);
  }
});
```

And change the import from `import { test } from '@jupyterlab/galata';` to `import { test } from './fixtures';` in the test suites.

This works well. However maybe there is common condition that could be used for most lab-based applications?

### Additional context

Switching to Galata in RetroLab: https://github.com/jupyterlab/retrolab/pull/273

**Comment by fcollonval** [comment]
This one is really tricky - waiting for the application readiness is quite complex. And the potential solution would be to test for some attribute on the exposed `window.jupyterapp` object. But we need to be sure extension authors won't be confused it with something like `isReady` as they should not wait for it to carry out action. The other point is that it may not be self contained; like the current function for JupyterLab is strongly coupled with the `launcher-extension` package. So depending on the evolution of the code base, the criteria to switch the application attribute may change.

**Comment by jtpio** [comment]
Right. Maybe the best for now would be to document how to override the fixture like in RetroLab.
